toolchain/sem_ir/name_scope.h

```diff
@@ -18,13 +18,104 @@ enum class AccessKind : int8_t {
   Private,
 };
 
+// Represents the result of a name lookup.
+// A lookup result can be:
+// * Successful returning a valid `InstId`,
+// * Not found when the name wasn't declared or poisoned.
+// * Poisoned when the name wasn't declared but was poisoned.
+// * Represent that an error has occurred during lookup. This is still
+// considered found and the error `InstId` is considered valid.
```

Comment:
What do you think of indenting wrapped bullets? I find it can help make clear where bullets start and end. i.e.:

```suggestion
//   considered found and the error `InstId` is considered valid.
```

(similar below)

Reply:
Done.